LeetCodeSolutions/Dota2Senate/Dota2Senate.cs

```diff
@@ -0,0 +1,140 @@
+namespace LeetCodeSolutions;
+public class Dota2SenateSolution {
+    private enum Party{
+        Dire,
+        Radiant
+    }
+
+    private enum SenatorAction{
+        AnnounceVictory,
+        BanOpponent
+    }
+
+
+    private class Senator{
+        private readonly Party party;
+        public Senator(char initial){
+            switch(initial){
+                case 'D':
+                    party = Party.Dire;
+                    break;
+                case 'R':
+                    party = Party.Radiant;
+                    break;
+            }
+        }
+
+        
+        public SenatorAction ExerciseRight(Dictionary<Party, int> activeSenateCount){
+            if(activeSenateCount[party] > 0 && activeSenateCount[OpposingParty()] == 0){
+                return SenatorAction.AnnounceVictory;
+            }else{
+                return SenatorAction.BanOpponent;
+            }
+        }
+
+        public Party OpposingParty(){
+            switch(party){
+                case Party.Dire:
+                    return Party.Radiant;
+                case Party.Radiant:
+                    return Party.Dire;
+                default:
+                    return Party.Dire;
+            }
+        }
+
+        public Party Party => party;
+    }
+
+    private class Senate{
+        private readonly Dictionary<Party, int> senatorCount;
+        private readonly Dictionary<Party, int> partyVetos;
+        private Queue<Senator> activeSenators;
+        private bool victoryDeclared;
+        private Party winningParty;
+        public Senate(string senators){
+            senatorCount = IntializeDictionary();
+            partyVetos = IntializeDictionary();
+            activeSenators = new Queue<Senator>();
+            victoryDeclared = false;
+
+            foreach(char partyInitial in senators){
+                AddSenator(partyInitial);
+            }
+        }
+
+        private void AddSenator(char initial){
+            Senator senator = new(initial);
+            activeSenators.Enqueue(senator);
+            senatorCount[senator.Party]++;
+        }
+        private static Dictionary<Party, int> IntializeDictionary(){
```

Comment:
might look into a plugin that lints c# code. The inconsistency of spacing here isn't an issue thanks to small units, but it's still annoying.